api/service.go

```diff
@@ -505,3 +509,25 @@ func contextsForServiceProvision(s *service.Service) []permTypes.PermissionConte
 		permission.Context(permTypes.CtxService, s.Name),
 	)
 }
+
+func filterServicesInstancesByTags(sInstances []service.ServiceInstance, tags []string) []service.ServiceInstance {
+	if len(tags) == 0 {
+		return sInstances
+	}
+
+	tagMap := make(map[string]struct{}, len(tags))
+	for _, tag := range tags {
+		tagMap[tag] = struct{}{}
+	}
+
+	var filteredInstances []service.ServiceInstance
```

Comment:
for performace reasons you may initialize a slice with high capacity like this:
```suggest
filteredInstances := make([]service.ServiceInstance, 0, len(sInstances))
```

Reply:
excellent tip fo perfomace, thanks!